Rework communication folders

Elydo-dev opened the pull request:
Improve architecture to match client side :
- Create impl directory
- Change imports path

Elydo-dev committed:
Refactor directories / imports to implementation files

AmauryMichel committed:
Prevent the user from logging in if they're already connected (#48)

* Check if user is already connected

* handle error if sending 'user connect error'

---------

Co-authored-by: ilianAZZ <ilian572@gmail.com>

ilianAZZ committed:
Merge branch 'dev' into dev-rework-folders

ilianAZZ reviewed (changes requested):
a "ComService" interface must be created 
please, copy client-side comm module

Elydo-dev committed:
Rework architecture :
- Create ComNotifierServices and ComServices interface
- Create ComServicesProvider class
- Refactor usages for using interfaces

Elydo-dev committed:
Fix port server

Elydo-dev committed:
Merge remote-tracking branch 'origin/dev-rework-folders' into dev-rework-folders

Elydo-dev committed:
Merge dev conflicts

Elydo-dev committed:
Merge remote-tracking branch 'origin/dev' into dev

# Conflicts:
#	src/main/java/fr/utc/onzzer/server/communication/impl/ServerCommunicationController.java

ilianAZZ commented:
Please, resolve conflicts

Elydo-dev committed:
Merge branch 'dev' into dev-rework-folders

# Conflicts:
#	src/main/java/fr/utc/onzzer/server/communication/impl/ServerCommunicationController.java

Elydo-dev committed:
Fix port server

Elydo-dev committed:
Merge remote-tracking branch 'origin/dev' into dev-rework-folders

Elydo-dev committed:
Fix handling rating
